src/v1/RequirementsChecker.php

```diff
@@ -130,21 +130,48 @@ private function check() {
 					$exists  = defined( 'EDD_VERSION' );
 					break;
 				default:
-					$version = false;
+					$version = isset( $properties['current'] ) ? $properties['current'] : false;
+					$exists  = isset( $properties['exists'] ) ? $properties['exists'] : false;
+
+					if ( is_callable( $exists ) ) {
+						$exists = call_user_func( $exists );
+					}
 					break;
 			}
 
 			if ( ! empty( $version ) ) {
 				$this->requirements[ $requirement_id ] = array_merge( $this->requirements[ $requirement_id ], array(
 					'current' => $version,
 					'checked' => true,
-					'met'     => version_compare( $version, $properties['minimum'], '>=' ),
-					'exists'  => isset( $exists ) ? $exists : $this->requirements[ $requirement_id ]['exists']
+					'met'     => $this->minimumVersionMet( $version, $properties['minimum'] ),
+					'exists'  => (bool) $exists,
 				) );
 			}
 		}
 	}
 
+	/**
+	 * Determines if the minimum version has been met.
+	 *
+	 * @since 1.0.1
+	 *
+	 * @param string|\Closure $currentVersion
+	 * @param string          $minimumVersion
+	 *
+	 * @return bool
+	 */
+	private function minimumVersionMet( $currentVersion, $minimumVersion ) {
```

Comment:
Putting my comment here but not sure if the adjustment would be in this method or above. I wanted to see if I could break the implementation because, well, because. So I thought I would try seeing what happens if I wanted to just require a specific plugin, and not a specific version, like this:

```php
	$requirements['mailpoet'] = array(
		'name'    => 'MailPoet',
		'exists'  => static function () {
			return defined( 'MAILPOET_VERSION' );
		},
	);
```

Unlikely, but it could happen, if I was sloppy. In this case, the checks fail and the EDD MailPoet extension is not fully activated:

![image](https://user-images.githubusercontent.com/4853066/144620302-a57b13d3-a64e-492c-a349-f9694fca9cb9.png)

Reply:
I might want to split that out into another issue.

We default the "minimum" version to `1`. I'm not sure I like defaulting the "current" version to `1` because that's quite an assumption.

So if this is something we want to support I guess I might just want to spend more time thinking about how best to implement it. (The quick fix, of course, would be to just set `'met'` to `true` if the minimum/current are omitted, but it does exist. But I'm not sure if I want to do that.)